Ensure OS_UNFAIR_LOCK_INLINE=1 is defined throughout the WebKit project.

MenloDorian opened the pull request:
#### 215ff6dcf112d3d029c596773ff1f5e8c4bdfa76
<pre>
Ensure OS_UNFAIR_LOCK_INLINE=1 is defined throughout the WebKit project.
<a href="https://bugs.webkit.org/show_bug.cgi?id=286268">https://bugs.webkit.org/show_bug.cgi?id=286268</a>
<a href="https://rdar.apple.com/133813123">rdar://133813123</a>

Reviewed by Yusuke Suzuki.

Apple port&apos;s libpas relies on this being consistently defined.
Now, we can remove the workarounds in ARKitSPI.h and WebGPU.xcconfig.

* Configurations/CommonBase.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig:
* Source/WebCore/PAL/pal/spi/cocoa/ARKitSPI.h:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/bmalloc/libpas/src/libpas/pas_lock.h:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:

Canonical link: <a href="https://commits.webkit.org/289181@main">https://commits.webkit.org/289181@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5b15363b1030b4d85099b4623fffda16f75620c4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85608 "4 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5301 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/40011 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90696 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36608 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/87658 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5501 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13281 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66527 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24336 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88620 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4229 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77734 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46810 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4098 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31997 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35679 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/78617 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74759 "Found 1 new API test failure: TestWebKitAPI.ResourceLoadStatistics.MigrateDistinctDataFromTableWithMissingIndexes (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32853 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/92361 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/84606 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12929 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9484 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75258 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13152 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73566 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74395 "Found 101 new API test failures: /TestWebKit:WebKit2UserMessageRoundTripTest.WKURLRequestRef, /TestWebKit:WebKit.OnDeviceChangeCrash, /TestWebKit:WebKit.PageLoadState, /WebKitGTK/TestGeolocationManager:/webkit/WebKitGeolocationManager/current-position, /TestWebKit:WebKit.InjectedBundleBasic, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/create-navigation-data, /TestWebKit:WebKit.GeolocationBasic, /TestWebKit:WebKit2.ProvisionalURLAfterWillSendRequestCallback, /WebKitGTK/TestInputMethodContext:/webkit/WebKitInputMethodContext/invalid-sequence, /WebKitGTK/TestUIClient:/webkit/WebKitWebView/usermedia-enumeratedevices-permission-check ... (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18629 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17064 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/5042 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13348 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12946 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18325 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/106996 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12745 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25791 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16178 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14528 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
In `pas_lock.h`, let's add check like

```
#ifndef OS_UNFAIR_LOCK_INLINE
#error "os_unfair_lock_inline needs to be enabled"
#endif
```

For Internal SDK build.

Constellation commented on Source/bmalloc/libpas/src/libpas/pas_lock.h:
```diff
@@ -117,11 +117,12 @@ PAS_END_EXTERN_C;
-#ifndef OS_UNFAIR_LOCK_INLINE
-#define OS_UNFAIR_LOCK_INLINE 1
+#if !defined(OS_UNFAIR_LOCK_INLINE) || !OS_UNFAIR_LOCK_INLINE
+#error OS_UNFAIR_LOCK_INLINE needs to be enabled.
```
This needs to be quoted, `"OS_UNFAIR_LOCK_INLINE needs to be enabled."`.

Commit:
Ensure OS_UNFAIR_LOCK_INLINE=1 is defined throughout the WebKit project.
https://bugs.webkit.org/show_bug.cgi?id=286268
rdar://133813123

Reviewed by Yusuke Suzuki.

Apple port's libpas relies on this being consistently defined.
Now, we can remove the workarounds in ARKitSPI.h and WebGPU.xcconfig.

* Configurations/CommonBase.xcconfig:
* Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig:
* Source/WebCore/PAL/pal/spi/cocoa/ARKitSPI.h:
* Source/WebGPU/Configurations/Base.xcconfig:
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/bmalloc/libpas/src/libpas/pas_lock.h:
* Tools/DumpRenderTree/mac/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/WebKitTestRunner/Configurations/Base.xcconfig:

Canonical link: https://commits.webkit.org/289181@main
